### PR TITLE
Add user car page

### LINKED ISSUE #17: Add User Car page
As a user, from the calculate car footprint page, when I click on the 'Add Car' link, 
I am taken to a new car form page where I see fields for: 
- [ ] make
- [ ] model
- [ ] year
- [ ] mpg
- [ ] fuel_type

When I submit the form, I am redirected to the calculate car footprint page

### PR DESCRIPTION
**Pull request by Jonathan-M-Wilson**
Add functionality allowing an authenticated user to visit a form to add a new car. 

This pr is a stepping stone for front-end issue 18. This is the basic functionality, without any graphql implementations. 

## Description

- Created a simple spec for the car monthly mileages page (Carbon Footprint Calculation for Car Trip wireframe). 

- Added associated controller and view for spec, the view is empty other than a link to add a new car.

- Created spec for the add a new car form page (Add Car wireframe) 

- Added associated controller. This controller is unfinished, check the notes in the car_controller file. 

- Created a view with forms for adding a new car.

## Motivation and Context

Closes #17 
Adds navigation to both the car monthly mileages page, and creates the adding a new car form. 

## How Has This Been Tested?

Specs for both the car monthly mileages page and for adding a new car form page have been added. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have updated the README to reflect any changes.

### CODE REVIEW

File: app/controllers/car_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CarController < ApplicationController
+  def new
+    @user = current_user
+  end
+
+  def create
+    # TODO
+    # before redirect, send params to facade/service to make a graphql mutation query
+    # to save car into backend db
+    # respond with new car data
```
Comment (c-bartell) [inline app/controllers/car_controller.rb:10]: Can we build this out with a mocked out spec that takes the query and returns the expected data?

File: config/routes.rb [hunk 2]
```diff
@@ -5,4 +5,8 @@
 
   get 'auth/google_oauth2/callback', to: 'sessions#create'
   get '/dashboard', to: 'dashboard#show'
+
+  get '/car_monthly_mileages/new', to: 'car_monthly_mileages#new'
+  get '/cars/new', to: 'car#new'
+  post '/cars', to: 'car#create'
```
Comment (carson-jardine) [inline config/routes.rb:11]: Can we use `resources` for this?

**Review by carson-jardine — APPROVED** [review]
Looks awesome! I think we can make the routes look nicer by using `resources` but other than that 👍

File: config/routes.rb (hunk 2 above)
Comment (c-bartell) [inline config/routes.rb:11]: We could refactor this with `resources` syntax.

**Review by c-bartell — APPROVED** [review]
Overall, I think this looks really good!

File: app/controllers/car_controller.rb (hunk 1 above)
Comment (jdcarey128) [reply app/controllers/car_controller.rb:10]: This lays the ground work, but story #18 should build on this functionality. This is an example where we didn't do a great job with grooming the stories beforehand.

**Review by jdcarey128 — APPROVED** [review]

**Pull request merged**